rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,225 @@
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Is the list of installation methods correct?
+I took it from my personal history as well as the NixOS website, but there may be others.
+Also, I may have the wrong job name, as I tried to guess the correct job name from the various links.
+
+Do we need a specific `redistributableWhenPatched` field on the license?
+It feels like this would be a bit too much, and probably `redistributable` would be enough.
+However, we may need to have it still.
+
+Will we need to redistribute some derivations with `runnableOnHydra = false`?
+For example, some firmware might not be legal to run on hydra.
+However, Hydra will never actually try to run it, as it cannot be used at runtime to build other packages.
+Maybe even `runnableOnHydra` could be better named to encompass this case too?
+
+# Future work
+[future]: #future-work
+
+- **Actually tagging licenses and packages as `runnableOnHydra`.**
+  Without this, this RFC would have no impact.
+  This will be done package-by-package, and should require no RFC, unless there are significant disagreements on whether a license should be runnable on hydra or not.
+
+- **Monitoring Hydra to confirm it does not push too much data to S3.**
+  If this change causes Hydra to push an economically non-viable amount of data to S3, then we should revert the addition of `runnableOnHydra` to the relevant packages and reconsider.
+
+- **Culling NVidia kernels and CUDA derivations.**
+  We suggest not caring too much about S3 size increases in the first step, considering the numbers from the resolved questions section.
+  However, if compression is less efficient than could be expected, we could be required to cull old NVidia kernels and/or CUDA derivations.
+  This would reduce the availability of older or more niche configurations, in exchange with reducing Hydra closure size.
+  Or we could move them to a set in which Hydra does not recurse.
+  For now, this is left as future work, that should be handled close to tagging the relevant derivations as `runnableOnHydra`.
+
+- **Modifying Hydra to allow building and redistributing packages that it is not legally allowed to run.**
+  This would be a follow-up project that is definitely not covered by this RFC due to its complexity, and would require a new RFC before implementation.
+
+- **Validating licenses and dependencies.**
+  We may be interested in figuring out the aggregate license of one derivation.
+  This could be automatically computed by evaluating the Nix scripts.
+  In particular, we could have a specific `enforceFree` meta argument that'd enforce that this derivation as well as all dependencies are transitively free.
+  Implementing this may be doable in pure nix, or could require an additional hydra check.
+  This is left as future work, because even without validating licenses this RFC probably reduces the risk for FOSS users from installing proprietary software.
```

Comment:
I really do not see how it reduces that risk at all and think this line should be removed.

Reply:
See the MacOS discussion above.